UI/script.js

```diff
@@ -1,79 +1,450 @@
+        if (item === tag) {
+          num += 1;
         }
+      });
+    });
+    return num === configTags.length;
+  }
+
+  addAll(posts) {
+    const notValid = [];
+    posts.forEach((post) => {
+      if (this.addPhotoPost(post) === false) {
+        notValid.push(post);
+      }
+    });
+    return notValid;
+  }
+
+  getPhotoPosts(skip = 0, count = 10, filterConfig = PostModel._DEFAULT_FILTER) {
+    const filtratedPosts = this._photoPosts.filter(
+      post => post.createdAt.getTime() >= filterConfig.dateFrom.getTime()
+        && post.createdAt.getTime() <= filterConfig.dateTo.getTime()
+        && (post.author === filterConfig.authorName
+          || filterConfig.authorName === '')
+        && (this._isIntersect(post.hashtags, filterConfig.hashtags)
+          || filterConfig.hashtags.length === 0),
+    ).sort((a, b) => b.createdAt.getTime() - a.createdAt.getTime());
+    const result = [];
+    let number = 0;
+    for (let i = skip; number < count && i < filtratedPosts.length; i += 1) {
+      result.push(filtratedPosts[i]);
+      number += 1;
     }
-    function getPhotoPost(id) {
-        var result;
-        postArray.forEach(function (item) {
-            if(item.id == id){
-                console.log("id " + id + " found ");
-                console.log(item);
-                result = item;
-            }
-        });
-        if (result != undefined) {
-            return result;
-        }
-        console.log("id " + id + " is not found.");
+    return result;
+  }
+
+  getPhotoPost(id) {
+    let result;
+    this._photoPosts.forEach((item) => {
+      if (item.id === id) {
+        result = item;
+      }
+    });
+    if (result !== undefined) {
+      return result;
     }
-    function removePhotoPost(id) {
-        var index = -1;
-        if(getPhotoPost(id)) {
-            index = getPhotoPost(id).id;
-        }
-        if (index != -1) {
-            postArray.splice(index, 1);
-            console.log("Post with id " + id + " deleted.");
-            return true;
-        }
-        else {
-            console.log("Post with id " + id + " not found.");
-            return false;
+    return result;
+  }
+
+  _validateChangeableFields(post) {
+    if (!post.hasOwnProperty('description') || post.description.length >= 200 || typeof post.description !== 'string'
+      || !post.hasOwnProperty('photoLink') || typeof post.photoLink !== 'string' || post.photoLink === ''
+      || !post.hasOwnProperty('hashtags')) {
+      return false;
+    }
+    return true;
+  }
+
+  _validateUnChangeableFields(post) {
+    if (!post.hasOwnProperty('id') || typeof post.id !== 'string'
+      || !post.hasOwnProperty('createdAt')
+      || !post.hasOwnProperty('author') || typeof post.author !== 'string' || post.author === ''
+      || !post.hasOwnProperty('likes')) {
+      return false;
+    }
+    return true;
+  }
+
+  _validatePhotoPost(post) {
+    if (this._validateUnChangeableFields(post) === false
```

Comment:
Do not compare Boolean it's ready for boolean algebra as it is.

return this._validateUnChangeableFields(post) && this._validateChangeableFields(post);